packages/core/src/tools/serialisation/contextManager.ts

```diff
@@ -51,8 +52,13 @@ export function createContextManager(customerDataType: CustomerDataType, compute
     getContext: () => deepClone(context),
 
     setContext: (newContext: Context) => {
-      context = sanitize(newContext)
-      computeBytesCountThrottled(context)
+      if (getType(newContext) === 'object') {
+        context = sanitize(newContext)
+        computeBytesCountThrottled(context)
+      } else {
+        context = {}
+        bytesCountCache = 0
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Perhaps we can merge this code with `clearContext`